pkg/cover/backend/elf.go

```diff
@@ -417,23 +424,65 @@ func readCoverPoints(file *elf.File, info *symbolInfo) ([2][]uint64, error) {
 	if err != nil {
 		return pcs, fmt.Errorf("failed to read .text: %v", err)
 	}
-	end := len(data) - callLen + 1
+	end := len(data)
+	if arch == targets.AMD64 {
+		end = end - callLen + 1
+	}
 	for i := 0; i < end; i++ {
-		pos := bytes.IndexByte(data[i:end], 0xe8)
+		pos := bytes.IndexByte(data[i:end], opcode)
 		if pos == -1 {
 			break
 		}
 		pos += i
 		i = pos
-		off := uint64(int64(int32(binary.LittleEndian.Uint32(data[pos+1:]))))
 		pc := text.Addr + uint64(pos)
-		target := pc + off + callLen
+		var off uint64
+		var target uint64
+		if arch == targets.AMD64 {
+			off = uint64(int64(int32(binary.LittleEndian.Uint32(data[pos+1:]))))
+			target = pc + off + uint64(callLen)
+		} else if arch == targets.ARM64 {
+			b := data[pos-callLen+1 : pos]
+			b = append(b, 0x00)
+			off = uint64(int64(int32(binary.LittleEndian.Uint32(b))))
+			off = 4 * off
+			target = pc + off - uint64(callLen) + 1
+			pc -= 3
+		}
 		if target == info.tracePC {
 			pcs[0] = append(pcs[0], pc)
 		} else if info.traceCmp[target] {
 			pcs[1] = append(pcs[1], pc)
 		}
 	}
+	if arch == targets.ARM64 {
+		opcode = 0x97
```

Comment:
This is nasty...
Ideally we use something like bytes.IndexAny to find all opcodes in one pass, but unfortunately IndexAny operates on runes rather than bytes:
https://golang.org/pkg/bytes/#IndexAny
Thinking what to do with this...

Reply:
OK, I think this would be a reasonable way to express what we want:
```
opcodes := [2]byte{0xe8, 0xe8}
if arch == targets.ARM64 {
	opcodes = [2]byte{0x94, 0x97}
}
for i, b := range data {
	if b != opcodes[0] && b != opcodes[1] {
		continue
	}
	...
}
```
And then we will need an if for opcode when we calculate arm64 target.